### PR TITLE
Fix infinite recursion for template intersection type accepts()

### PR DESCRIPTION
**Pull request by rvanvelzen**
Fixes phpstan/phpstan#7500

### CODE REVIEW

File: src/Type/IntersectionType.php [hunk 1]
```diff
@@ -144,10 +143,6 @@ public function isSubTypeOf(Type $otherType): TrinaryLogic
 
 	public function isAcceptedBy(Type $acceptingType, bool $strictTypes): TrinaryLogic
 	{
-		if ($acceptingType instanceof self || ($acceptingType instanceof UnionType && !$acceptingType instanceof TemplateUnionType)) {
-			return $acceptingType->accepts($this, $strictTypes);
-		}
-
```
Comment (rvanvelzen) [inline src/Type/IntersectionType.php:150]: Nothing appears to break by removing this :man_shrugging:

**Comment by ondrejmirtes** [comment]
Thank you! Removing code is always nice 😊

**Comment by VincentLanglet** [comment]
@rvanvelzen @ondrejmirtes Am I wrong or the testBug7500 shouldn't report any error ?

https://phpstan.org/r/29203f2e-5b6c-46bf-b91a-baf7d3839591 works fine
but https://phpstan.org/r/05a8098a-1a57-460c-8707-3027e65605da doesn't.

```
\PHPStan\dumpType($res[$position]);
```
is `PositionEntityInterface&TgEntityInterface` but since
```
/** @phpstan-var array<T> $res */
$res = [];
``` 
it should be `T of PositionEntityInterface&TgEntityInterface` IMHO.

Should I open an issue ?

**Comment by ondrejmirtes** [comment]
@VincentLanglet Comments on closed PRs tend to get lost, please open a new bug report about this, thanks.

**Comment by VincentLanglet** [comment]
Yes I did it @ondrejmirtes  https://github.com/phpstan/phpstan/issues/7511 with a simpler reproducer

**Pull request merged**